### PR TITLE
Merge RustLSP vscode types here

### PR DESCRIPTION
**Pull request by bruno-medeiros**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -975,4 +1317,4 @@ pub struct RenameParams {
     /// appropriate message set.
     #[serde(rename="newName")]
     pub new_name: String,
-}
+}
```
Comment (Marwes) [inline src/lib.rs]: Could you add a newline to the end of the file? It makes diffs later on cleaner.

**Comment by Marwes** [comment]
Nice!. I'd merge this but what do you think of https://github.com/jonathandturner/rustls/issues/52#issuecomment-256147815? I'd prefer to not need the extra `Option`. Also sent you merge access.

**Comment by bruno-medeiros** [comment]
Hum, I don't seem to be able to merge the PR, says I'm not authorized, even though I just accept your invitation.

**Comment by Marwes** [comment]
It says you have write access on the repository page so that is very strange.

**Comment by Marwes** [comment]
I can merge this if you want but it would be nice to check that you can actually merge things.

**Comment by bruno-medeiros** [comment]
I can't merge. The page here says to me: 
"This pull request can be automatically merged by project collaborators
Only those with write access to this repository can merge pull requests."

I can't push directly from git, I get an error about protected branches? Maybe that's the underlying issue?
https://github.com/blog/2051-protected-branches-and-required-status-checks

**Comment by Marwes** [comment]
master is a protected branch (from me as well) since I think its good to at least need a PR for changes.

**Comment by Marwes** [comment]
Removed you as collaborator and resent it.

**Comment by Marwes** [comment]
Or maybe it is because this PR was created before you got the invite/accepted it?

**Comment by bruno-medeiros** [comment]
> Or maybe it is because this PR was created before you got the invite/accepted it?

Re-accepted, but still can't merge. I doubt it's because of the above. It must be because master is protected. If you set it up so that changes have to be approved, I suspect another collaborator must approve the changes, even if I am a collaborator now as well. (I'm don't well how protected branches, but I suspect its so)

**Comment by Marwes** [comment]
Now maybe?

**Comment by bruno-medeiros** [comment]
now it worked (y)

**Comment by Marwes** [comment]
🎉 It was a setting in protected branches which I hadn't seen before. Might have been there by default or I ticked it by mistake.

```
Restrict who can push to this branch Specify people or teams allowed to push to this branch. Required status checks will still prevent these people from merging if the checks fail. 
```

**Comment by bruno-medeiros** [comment]
I just tried a push directly from Git as well (version number bump)

**Comment by Marwes** [comment]
Ok, I can publish in an hour or so, can give you permission to publish then as well.

**Pull request merged**